### ISSUE TITLE
Error when adding bundle as a single component instead of silently failing

### ISSUE THREAD

**Issue by WesterWest** [bevyengine/bevy#392]
When you want to add `*Components`, you must use `.spawn(*Components)` and not `.spawn((*Components, ..))` the spawning will fail, but there is no indication to the user, that it won't work.

This is also the case when adding `.with(*Components)` instead of `.with_bundle(*Components)`

This should definitely tell user, that it's not correct to add bundles this way and not silently fail but compile no problem.

**Comment by DJMcNab** [comment]
I wonder if it would be possible to cheat a little bit, by making the default Bundles `!Sync`, maybe by adding a `PhantomData<*const ()>` field?
Obviously we'd have to change the `macros` crate to add `#[bundle(ignore)]`.
I'd probably also add `#[bundle(optional)]` at the same time.
I'm going to do some investigative work around this soon.

**Comment by CleanCut** [comment]
I wasted an hour of my life on this today (again).  Can you spot my error?

```rust
struct Something;

fn setup(commands: &mut Commands, asset_server: Res<AssetServer>) {
    commands
        .spawn(Camera2dBundle::default())
        .spawn((Something,))
        .with(Text2dBundle {
            text: Text {
                value: "Some Text".to_string(),
                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
                style: TextStyle {
                    font_size: 60.0,
                    color: Color::WHITE,
                    alignment: TextAlignment {
                        vertical: VerticalAlign::Center,
                        horizontal: HorizontalAlign::Center,
                    },
                },
            },
            ..Default::default()
        });
}
```

Answer: `.with(...)` was silently failing.  It needed to be `.with_bundle(...)`

**Comment by DJMcNab** [comment]
I've had a new idea of how to 'fix' this, which would be to go super simple
```rust
fn illegal_component_system<T: Bundle>(q: Query(Entity, With<T>)){
for e in q.iter(){
 //...print message depending on settings
}
}
```
And then add a method on app to add this system

Possibly also add a field on App which would disable adding these

**Comment by cart** [comment]
> I've had a new idea of how to 'fix' this, which would be to go super simple

Its a bit heavy for my taste to prevent what in most cases is a "newbie error". And would require somehow connecting the World impl and the Schedule impl to ensure a new system is inserted for each Bundle. 

Haha its not a full solve, but we could consider un-implementing Bundle entirely for tuples. (ex: the only way to spawn something is through a pre-constructed Bundle derive, otherwise you must use commands.with())

It doesn't solve the `commands.with(SomeBundle)` problem though.

**Comment by alice-i-cecile** [comment]
As part of this, trying to access a bundle as a component (commonly as part of a query) should also fail or warn. See #403 for related work.

**Comment by alice-i-cecile** [comment]
Fixed by #2254!
